SkyEditor.RomEditor.Rtdx/Domain/Rtdx/Structures/Sir0StringList.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using SkyEditor.IO.Binary;
+using SkyEditor.RomEditor.Domain.Common.Structures;
+
+namespace SkyEditor.RomEditor.Domain.Rtdx.Structures
+{
+    public class Sir0StringList
+    {
+        public List<string> Entries { get; } = new List<string>();
+
+        private Encoding encoding;
+
+        public Sir0StringList(Encoding encoding)
+        {
+            this.encoding = encoding;
+        }
+
+        public Sir0StringList(byte[] data) : this(data, Encoding.Unicode)
+        {
+        }
+        
+        public Sir0StringList(byte[] data, Encoding encoding)
+        {
+            this.encoding = encoding;
+            
+            // IReadOnlyBinaryDataAccessorExtensions.ReadNullTerminatedString() doesn't work with multibyte strings
```

Comment:
There's IReadOnlyBinaryDataAccessorExtensions.ReadNullTerminatedUnicodeString, which works specifically for 2-byte characters, although it too won't work for 4-byte characters.